### PR TITLE
Support for Wildfly10 and RHEL7

### PR DESCRIPTION
**Pull request by therek**
Added support for Wildfly 10 and RHEL7.

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/coi-gov-pl/puppet-jboss/pull/95" target="_blank">ReviewNinja

File: examples/jboss/wildfly10.pp [hunk 1]
```diff
@@ -0,0 +1,4 @@
+class { 'jboss':
+  product => 'wildfly',
+  version => '10.0.0.Final',
```
Comment (Cosaquee) [inline examples/jboss/wildfly10.pp]: Most recent version is 10.1.0, so maybe test always for the most recent.

File: manifests/internal/compatibility/eap.pp [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+            fail("Unsupported OS: ${::operatingsystem}")
+          }
+        }
+      }
+      'Debian': {
+        case $::lsbdistcodename {
+          'lenny','squeeze','lucid','natty','wheezy','precise','quantal','raring','saucy','trusty','utopic' : {
+            $initsystem = 'SystemV'
+          }
+          'jessie','vivid','wily','xenial': {
```
Comment (Cosaquee) [inline manifests/internal/compatibility/eap.pp]: 16.04 -> yakkety, 17.04 -> zesty

File: manifests/internal/compatibility/wildfly.pp [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+      'RedHat': {
+        case $::operatingsystem {
+          'RedHat', 'CentOS', 'OracleLinux', 'Scientific', 'OEL': {
+            if (versioncmp($::operatingsystemrelease, '7.0') > 0) {
+              $initsystem = 'SystemD'
+            } else {
+              $initsystem = 'SystemV'
+              $versioned_name = "${::operatingsystem} ${::operatingsystemrelease}"
+              if jboss_to_bool($jboss::java_autoinstall) {
+                warning("Wildfly 10 requires Java 8 or greater, which is not suppoted by default on ${versioned_name}.")
```
Comment (Cosaquee) [inline manifests/internal/compatibility/wildfly.pp]: suppoted -> supported

File: manifests/internal/compatibility/wildfly.pp [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+            }
+          }
+          'Fedora': {
+            if (versioncmp($::operatingsystemrelease, '21') > 0) {
+              $initsystem = 'SystemD'
+            } else {
+              $initsystem = 'SystemV'
+              $versioned_name = "${::operatingsystem} ${::operatingsystemrelease}"
+              if jboss_to_bool($jboss::java_autoinstall) {
+                warning("Wildfly 10 requires Java 8 or greater, which is not suppoted by default on ${versioned_name}.")
```
Comment (Cosaquee) [inline manifests/internal/compatibility/wildfly.pp]: suppoted -> supported

File: manifests/internal/compatibility/wildfly.pp [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+            fail("Unsupported OS: ${::operatingsystem}")
+          }
+        }
+      }
+      'Debian': {
+        case $::lsbdistcodename {
+          'lenny','squeeze','lucid','natty','wheezy','precise','quantal','raring','saucy','trusty','utopic' : {
+            $initsystem = 'SystemV'
+          }
+          'jessie','vivid','wily','xenial': {
```
Comment (Cosaquee) [inline manifests/internal/compatibility/wildfly.pp]: 16.10 -> yakkety, 17.04 -> zesty

**Comment by cardil** [comment]
@Cosaquee @therek review my changes, please.

**Pull request merged**